### PR TITLE
Simplify conv_mkldnn op registration

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

OPs 
### Describe

Simplify conv_mkldnn op registration

Remove the custom type value using in conv mkldnn op kernel, preparing for migrating it to phi.

### CODE REVIEW

**Review by jiahy0825 — APPROVED** [review]
LGTM

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc [hunk 1]
```diff
@@ -828,48 +845,52 @@ class ConvMKLDNNOpKernel : public framework::OpKernel<T> {
...
-      auto bias_memory_p = handler.AcquireBiasMemoryWithReorder(bias, is_test);
-      args.insert({DNNL_ARG_BIAS, *bias_memory_p});
-    }
-
-    auto& astream = platform::MKLDNNDeviceContext::tls().get_stream();
-    conv_p->execute(astream, args);
-    astream.wait();
+    PD_VISIT_FLOAT_AND_INT8_TYPES(
+        filter->dtype(), "ConvMKLDNNHandlerT", ([&] {
+          ConvMKLDNNHandlerT<T, data_t, T_out> handler(
```
Comment (YuanRisheng) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:850]: 这里的处理逻辑看起来和原来的逻辑并不等价，原来T如果是float类型，K也必然是float，但这里看起来K的类型是根据fliter进行了选择

It seems that the processing logic here is not equivalent to the original logic. If T is a float type, K must also be a float, but it seems that the type of K is selected according to the filter

Comment (chenwhql) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:850]: The modified implementation covers the original implementation. The real requirement here is that the data type of filter may be inconsistent with x, and K is selected according to the data type of the filter, so the data type needs to be dispatched again in the kernel.

**Review by YuanRisheng — APPROVED** [review]

**Comment by chenwhql** [comment]
@jczaja @Silv3S Can you help to review this PR?

**Comment by chenwhql** [comment]
@jczaja @Silv3S Can you help to review this PR?

**Comment by jczaja** [comment]
@chenwhql  HI, I have just started evaluation of those changes on our internal system . Results should be by the end of week. We will review also this PR as soon our internal system confirms that all is fine.

**Comment by chenwhql** [comment]
> @chenwhql HI, I have just started evaluation of those changes on our internal system . Results should be by the end of week. We will review also this PR as soon our internal system confirms that all is fine.

@jczaja ok, thanks

**Review by Silv3S — APPROVED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@chenwhql  Hi, We have finished our CI runs and no problems was found with this PR.

File: paddle/fluid/operators/conv_op.cc [hunk 2]
```diff
@@ -229,31 +229,13 @@ framework::OpKernelType ConvOp::GetExpectedKernelType(
...
 #ifdef PADDLE_WITH_MKLDNN
   if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
-    int customized_type_value =
-        (input_data_type == framework::DataTypeTrait<int8_t>::DataType() ||
-         input_data_type == framework::DataTypeTrait<uint8_t>::DataType())
-            ? OperatorWithKernel::IndicateVarDataType(ctx, "Filter") ==
-                      framework::DataTypeTrait<int8_t>::DataType()
-                  ? kConvMKLDNNINT8WS8
-                  : kConvMKLDNNINT8
-            : kConvMKLDNNFP32;
```
Comment (sfraczek) [inline paddle/fluid/operators/conv_op.cc:239]: It seems then that those variables are declared in conv_op.h:33 but never used anymore

Comment (jiahy0825) [reply paddle/fluid/operators/conv_op.cc:239]: Thanks, I cleaned these variables by this PR: [[Opt Code] Clean conv_op useless variable](https://github.com/PaddlePaddle/Paddle/pull/47213)

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc (hunk 1 above)
Comment (sfraczek) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:850]: Why is `data_t` not named like `T` or `T_out` and what is it? How does it work now without need for `ops::kConvMKLDNNINT8WS8`?

**Pull request merged**